internal/core/metadata/v2/controller/http/deviceprofile_test.go

```diff
@@ -469,25 +492,159 @@ func TestAddDeviceProfileByYaml_MissingFile(t *testing.T) {
+	dic := mockDic()
+
+	controller := NewDeviceProfileController(dic)
+	assert.NotNil(t, controller)
+
+	deviceProfile := buildTestDeviceProfileRequest().Profile
+	noName := deviceProfile
+	noName.Name = ""
+	noDeviceResource := deviceProfile
+	noDeviceResource.DeviceResources = []dtos.DeviceResource{}
+	noDeviceResourceName := deviceProfile
+	noDeviceResourceName.DeviceResources = []dtos.DeviceResource{{
+		Description: TestDescription,
+		Tag:         TestTag,
+		Attributes:  testAttributes,
+		Properties: dtos.PropertyValue{
+			Type:      "INT16",
+			ReadWrite: "RW",
+		},
+	}}
+	noDeviceResourcePropertyType := deviceProfile
+	noDeviceResourcePropertyType.DeviceResources = []dtos.DeviceResource{{
+		Name:        TestDeviceResourceName,
+		Description: TestDescription,
+		Tag:         TestTag,
+		Attributes:  testAttributes,
+		Properties: dtos.PropertyValue{
+			ReadWrite: "RW",
+		},
+	}}
+	noCommandName := deviceProfile
+	noCommandName.CoreCommands = []dtos.Command{{
+		Get: true,
+		Put: true,
+	}}
+	noCommandGet := deviceProfile
+	noCommandGet.CoreCommands = []dtos.Command{{
+		Name: TestProfileResourceName,
+		Get:  false,
+	}}
+	noCommandPut := deviceProfile
+	noCommandPut.CoreCommands = []dtos.Command{{
+		Name: TestProfileResourceName,
+		Put:  false,
+	}}
+	tests := []struct {
+		name    string
+		Request dtos.DeviceProfile
+	}{
+		{"Invalid - No name", noName},
+		{"Invalid - No deviceResource", noDeviceResource},
+		{"Invalid - No deviceResource name", noDeviceResourceName},
+		{"Invalid - No deviceResource property type", noDeviceResourcePropertyType},
+		{"Invalid - No command name", noCommandName},
+		{"Invalid - No command Get", noCommandGet},
+		{"Invalid - No command Put", noCommandPut},
 	}
-	boundary := writer.Boundary()
+	for _, testCase := range tests {
+		t.Run(testCase.name, func(t *testing.T) {
+			valid, err := yaml.Marshal(testCase.Request)
+			require.NoError(t, err)
+			req, err := createDeviceProfileRequestWithFile(valid)
+			require.NoError(t, err)
 
-	err = writer.Close()
-	if err != err {
-		return nil, err
+			// Act
+			recorder := httptest.NewRecorder()
+			handler := http.HandlerFunc(controller.UpdateDeviceProfileByYaml)
+			handler.ServeHTTP(recorder, req)
+			var res common.BaseWithIdResponse
+			err = json.Unmarshal(recorder.Body.Bytes(), &res)
+
+			// Assert
+			assert.Equal(t, http.StatusBadRequest, recorder.Result().StatusCode, "HTTP status code not as expected")
+			assert.Equal(t, contractsV2.ApiVersion, res.ApiVersion, "API Version not as expected")
+			assert.Equal(t, http.StatusBadRequest, res.StatusCode, "HTTP status code not as expected")
+			assert.NotEmpty(t, string(recorder.Body.Bytes()), "Message is empty")
+		})
 	}
+}
 
-	req, _ := http.NewRequest(http.MethodPost, contractsV2.ApiDeviceProfileRoute+"/uploadfile", body)
-	req.Header.Set(clients.ContentType, "multipart/form-data; boundary="+boundary)
-	return req, nil
+func TestUpdateDeviceProfileByYaml_NotFound(t *testing.T) {
```

Comment:
Add to table drive. Use property to determine which mock of `UpdateDeviceProfile` to use.

Reply:
Added to table drive test.